runtime/compiler/z/codegen/J9BCDTreeEvaluator.cpp

```diff
@@ -4569,11 +4565,10 @@ J9::Z::TreeEvaluator::pdstoreVectorEvaluatorHelper(TR::Node *node, TR::CodeGener
       traceMsg(comp,"generating VSTRL for pdstore node->size = %d.\n", node->getSize());
       }
 
-   // No need to evaluate the address node of the pdstorei.
-   // generateVSIInstruction() API will call separateIndexRegister() to separate the index
-   // register by emitting an LA instruction. If there's a need for large displacement adjustment,
-   // LAY will be emitted instead.
-   TR::MemoryReference * targetMR = TR::MemoryReference::create(cg, node);;
+   // No need to evaluate the address node (first child) of the pdstorei.
```

Comment:
I think originally the second comment was meant to be used for `generateVSIInstruction` , it does not meant to say that it will evaluate the address node. As comment says, the VSI instruction generation will separate out the index register from the memory reference. Your current comment enhances what we put in first place, but we should also keep the second comment (May be before generate*).

Reply:
The call to create memory reference in those 3 evaluators goes through [1] which doesn't create an index register. Index register doesn't get generated on the path to `separateIndexRegister(...)` either so that's why the comment about `generateVSIInstruction` didn't feel relevant here.

[1] https://github.com/eclipse/omr/blob/26e52210d25d5b3cd84b1573d4060ef7b663303e/compiler/z/codegen/OMRMemoryReference.cpp#L470 "Memory Reference canUseRX path"
[2] https://github.com/eclipse/omr/blob/310f0efc871923145d85dd1f450b6a08dd11ef35/compiler/z/codegen/OMRMemoryReference.cpp#L2265 "separateIndexRegister(...)"

Reply:
`generateVSIInstruction` calls to `enforceSSFormatLimits` on the memory reference, which takes care of translating memory reference that needs all three `index register` , `base register` and `displacement` to the SS format memory reference generating necessary `LA` or `LAY` instruction. 

> The call to create memory reference in those 3 evaluators goes through [1] which doesn't create an index register.

I do not think that is true. It is common constructor that is uses to generate memory reference for all instruction. If you take a look at the code that would be executed for indirect load or store [2], you will see that it first tries to generate `tryBaseIndexDispl` for the memory reference which would allocate base register, index register and set displacement value for the memory reference. 


[1]. https://github.com/eclipse/omr/blob/310f0efc871923145d85dd1f450b6a08dd11ef35/compiler/z/codegen/OMRMemoryReference.cpp#L2054-L2067
[2]. https://github.com/eclipse/omr/blob/310f0efc871923145d85dd1f450b6a08dd11ef35/compiler/z/codegen/OMRMemoryReference.cpp#L1610

Reply:
ah true true, you are right. I'll update the PR